mysql/datadog_checks/mysql/data/conf.yaml.example

```diff
@@ -21,230 +29,280 @@ instances:
-    #       SELECT foo,
-    #              COUNT(*)
-    #       FROM table.events
-    #       GROUP BY foo
-    #     columns:
-    #       # Columns without a name are ignored. Add additional columns to skip:
-    #       # - {}
-    #       - name: foo
-    #         type: tag
-    #       - name: event.total
-    #         type: gauge
-    #     tags:
-    #       - test:mysql
-
-    ## @param max_custom_queries - integer - optional - default: 20
+    ## A list of custom queries used to retrieve custom metrics.
+    #
+    custom_queries:
+
+      -
+        ## @param query - string - optional
+        ## The SQL query to execute. It can be a simple statement, or a multi-line script.
+        #
+        # query: <QUERY>
+
+        ## A list representing each column, ordered sequentially from left to right.
+        ## The number of columns must be equal to the number of columns returned in the `query`.
+        #
+        columns:
+
+          -
+            ## @param name - string - optional
+            ## The suffix to append to `mysql.` to form the full metric name.
+            ## If `type` is `tag`, this column is considered a tag and applied to every metric
+            ## collected by this query.
+            #
+            # name: <NAME>
+
+            ## @param type - string - optional
+            ## The submission method (gauge, monotonic_count, etc.).
+            ## Use the 'tag' type to tag each metric in the row with the name and value
+            ## of the item in this column.
+            ## Use the 'count' type to perform aggregation for queries that return multiple
+            ## rows with the same or no tags.
+            #
+            # type: <TYPE>
+
+        ## @param tags - list of strings - optional
+        ## A list of tags to apply to each metric.
+        #
+        # tags:
+        #   - <KEY_1>:<VALUE_1>
+        #   - <KEY_2>:<VALUE_2>
+
+    ## @param max_custom_queries - number - optional - default: 20
     ## Set the maximum number of custom queries to execute with this integration.
     ##
     ## WARNING: This only applies to the deprecated `queries` option.
     #
     # max_custom_queries: 20
 
-    ## @param queries - object - optional
-    ## Define custom queries to collect custom metrics on your MySQL
+    ## Define custom queries to collect custom metrics from your MySQL database.
+    ##
     ## See https://docs.datadoghq.com/integrations/faq/how-to-collect-metrics-from-custom-mysql-queries to learn more.
     ##
-    ## WARNING: This option is deprecated and will be removed in a future release. Use the `custom_queries` option instead.
+    ## WARNING: This option is deprecated and will be removed in a future release.
+    ## Use the `custom_queries` option instead.
     #
-    # queries:
-    #   - query: <QUERY>
-    #     metric: <METRIC_NAME>
-    #     tags:
-    #       - <METRIC_TAG_KEY>:<METRIC_TAG_VALUE>
-    #     type: <METRIC_TYPE>
-    #     field: <FIELD_NAME>
```

Comment:
Similar to `custom_queries`, IMO the previous way using an "example" block is easier to use and understand for user.